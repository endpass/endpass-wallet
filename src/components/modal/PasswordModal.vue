<template lang="html">
  <div
    class="password-modal"
    data-test="password-modal"
  >
    <v-modal @close="close">
      <template slot="header">
        {{ title }}
      </template>

      <div>
        <p class="subtitle">Please enter your wallet password to continue.</p>
        <v-form
          id="password-form"
          :isFormValid="isFormValid"
        >
          <slot />
          <v-password
            v-model="jsonKeystorePassword"
            name="jsonKeystorePassword"
            validator="required"
            data-vv-as="password"
            v-validate="'required|min:8'"
            :error="errors.first('jsonKeystorePassword')"
            placeholder="Your Wallet Password"
            required
            data-test="input-password"
            autofocus
            @input="handleInput"
          />
        </v-form>
      </div>
      <div
        slot="footer"
        class="buttons"
      >
        <v-button
          :loading="processingConfirmation"
          form="password-form"
          data-test="submit-password"
          class-name="is-primary is-medium"
          :disabled="!isFormValid"
          @click="confirm"
        >
          Confirm
        </v-button>
      </div>
    </v-modal>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import VModal from '@/components/ui/VModal';
import VForm from '@/components/ui/form/VForm.vue';
import VPassword from '@/components/ui/form/VPassword.vue';
import VButton from '@/components/ui/form/VButton.vue';
import formMixin from '@/mixins/form';

export default {
  name: 'PasswordModal',

  props: {
    title: {
      type: String,
      default: 'Please enter your wallet password to continue',
    },
  },

  data() {
    return {
      jsonKeystorePassword: '',
      processingConfirmation: false,
    };
  },
  methods: {
    ...mapActions('accounts', ['validatePassword']),

    async confirm() {
      const { jsonKeystorePassword: password } = this;

      try {
        this.processingConfirmation = true;

        await this.validatePassword(password);

        this.$emit('confirm', password);
      } catch (err) {
        this.errors.add({
          field: 'jsonKeystorePassword',
          msg: 'Password is invalid',
          id: 'wrongPassword',
        });
      } finally {
        this.processingConfirmation = false;
      }
    },

    close() {
      this.$emit('close');
    },

    handleInput() {
      this.errors.removeById('wrongPassword');
    },
  },
  mixins: [formMixin],
  components: {
    VModal,
    VForm,
    VPassword,
    VButton,
  },
};
</script>

<style lang="scss">
.passsword-modal {
}
</style>
