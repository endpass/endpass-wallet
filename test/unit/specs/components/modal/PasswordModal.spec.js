import Vuex from 'vuex';
import VeeValidate from 'vee-validate';
import { shallow, createLocalVue, mount } from '@vue/test-utils';

import PasswordModal from '@/components/modal/PasswordModal';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VeeValidate);

describe('PasswordModal', () => {
  let wrapper;
  let store;
  let options;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        accounts: {},
      },
    });

    options = {
      store,
      localVue,
      methods: {
        validatePassword: () => null,
      },
    };
  });

  describe('render', () => {
    beforeEach(() => {
      wrapper = shallow(PasswordModal, {
        ...options,
      });
    });

    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('PasswordModal');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
