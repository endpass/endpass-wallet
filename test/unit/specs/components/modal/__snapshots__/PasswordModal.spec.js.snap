// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordModal render should render component 1`] = `
<div
  class="password-modal"
  data-test="password-modal"
>
  <v-modal>
    <template>
      
      Please enter your wallet password to continue
    
    </template>
     
    <div>
      <p
        class="subtitle"
      >
        Please enter your wallet password to continue.
      </p>
       
      <v-form
        id="password-form"
      >
         
        <v-password
          aria-invalid="true"
          aria-required="true"
          autofocus=""
          data-test="input-password"
          data-vv-as="password"
          data-vv-name="jsonKeystorePassword"
          name="jsonKeystorePassword"
          placeholder="Your Wallet Password"
          required=""
          validator="required"
        />
      </v-form>
    </div>
     
    <div
      class="buttons"
      slot="footer"
    >
      <v-button
        class-name="is-primary is-medium"
        data-test="submit-password"
        disabled="true"
        form="password-form"
      >
        
        Confirm
      
      </v-button>
    </div>
  </v-modal>
</div>
`;
